### DISCUSSION TITLE
How to upgrade to v3 incrementally?

### DISCUSSION THREAD

**Discussion by sangdth** [chakra-ui/chakra-ui#9184] [Q&A]
We have a huge project which use Chakra 2 in the last 2 years. It grows into +700 components, we already have paid customers but do not have a luxury of a dedicated team to migrate. What would be a best practices to migrate to v3?

I spent the weekend trying to write some wrappers that serve as a bridge between old and new components, but when I reach the `Form` vs `Field`s, I am totally exhausted and really want to give up.

**Comment by GuilhermeAmado** [comment]
You can't.
We are going through the same issue currently.
We postponed the migration to Chakra v3 because of the many breaking changes and different ways of doing things.

The incomplete docs currently doesn't help either.

**Comment by anilanar** [comment]
We are also exploring this topic right now. The idea is to alias v3 to something else (like `@chakra-ui/v3`) and to use v2 and v3 at the same time. We are currently looking into whether if there's a way for css reset and global styles to not conflict with each other.

**Reply by vicvytruve** [reply]
Did you find a acceptable solution for this ? We are also looking to migrate but it's too much work for the team

**Reply by anilanar** [reply]
Not yet, but we will continue investigating. The next candidate solution is to fork Chakra v2 and to fix its css-reset in such a way that they only apply to v2 components.

Otherwise aliasing the module name works. The only problem (that we are aware of) that is not solvable is css-reset conflicts.

Edit: Actually there was 1 more problem. You cannot use Chakra v3 and Emotion's css prop at the same time. For that, we had to implement a custom emotion component. Can share if needed.

**Reply by anilanar** [reply]
I'm curious what @segunadebayo thinks about this.

**Reply by anilanar** [reply]
We are still actively working on this, but slowly. Here's our progress so far: https://github.com/userlike/chakra-v2/commits/v2-fork/

**Reply by spjpgrd** [reply]
You're doing some huge lifting here.

Just so other people don't try to go down this road, here's what I was trying.

I've been trying to get both to co-exist in the same project as well with package aliasing —

`package.json`
```json
{
 "dependencies": {
    "chakra-ui/react-v2": "npm:@chakra-ui/react@^2.8.0",
    "chakra-ui/react-v3": "npm:@chakra-ui/react@^3.15.0",
    "chakra-ui/skip-nav-v2": "npm:@chakra-ui/skip-nav@^2.1.0",
    "emotion/react-v2": "npm:@emotion/react@^11.0.0",
    "emotion/react-v3": "npm:@emotion/react@^11.11.1",
    "emotion/styled-v2": "npm:@emotion/styled@^11.0.0",
    "emotion/styled-v3": "npm:@emotion/styled@^11.11.0",
    "framer-motion-v2": "npm:framer-motion@^4.0.0",
    "framer-motion-v3": "npm:framer-motion@^10.16.1"
    }
}
```

But since since the `@charka-ui` folder has shared utilities, it really screws things up.

![image](https://github.com/user-attachments/assets/0a173185-cf9f-48b0-aeac-188f687a93a4)

So I tried to break out like this, but it doesn't work since this doesn't create two`@chakra-ui` directories with the necessary utilities, you just get the `react` part.

<img width="285" alt="image" src="https://github.com/user-attachments/assets/de7afea6-9a4f-4800-9d26-8c5b182883c5" />

`package.json`
```json
{
 "dependencies": {
    "chakra-ui-react-v2": "npm:@chakra-ui/react@^2.8.0",
    "chakra-ui-react-v3": "npm:@chakra-ui/react@^3.15.0",
    "chakra-ui-skip-nav-v2": "npm:@chakra-ui/skip-nav@^2.1.0",
    "emotion-react-v2": "npm:@emotion/react@^11.0.0",
    "emotion-react-v3": "npm:@emotion/react@^11.11.1",
    "emotion-styled-v2": "npm:@emotion/styled@^11.0.0",
    "emotion-styled-v3": "npm:@emotion/styled@^11.11.0",
    "framer-motion-v2": "npm:framer-motion@^4.0.0",
    "framer-motion-v3": "npm:framer-motion@^10.16.1"
    }
}
```
